src/serializer/ResidualHeapVisitor.js

```diff
@@ -161,6 +162,7 @@ export class ResidualHeapVisitor {
   environmentRecordIdAfterGlobalCode: number;
 
   globalEnvironmentRecord: GlobalEnvironmentRecord;
+  parentGenerator: void | Generator;
```

Comment:
As a general guideline, I'd like to not introduce additional state in the visitor.